cg/meta/archive/ddn_dataflow.py

```diff
@@ -213,7 +223,7 @@ def retrieve_folders(self, sources_and_destinations: List[MiriaFile]) -> bool:
         return job_id
 
     def create_transfer_request(
-        self, sources_and_destinations: List[MiriaFile], is_archiving_request: bool
+        self, miria_file_data: List[MiriaFile], is_archiving_request: bool
```

Comment:
ah I see it got changed here. I don't see why `create_transfer_request` has to know the specifics of what files they transfer, i.e. MiriaFile in this case.

Can't we make a structure where we handle the different File types i.e. `MiriaFile` and what might come in the future and pass that class here? Then these functions can be more general and not tied to one specific file type.

Reply:
@ChrOertlin Do you mean that we would have in ArchiveHandler a "create_transfer_request" which takes List[FileTransferData] as input, and that we implement it in the different ArchiveHandlers such as DDNDataFlowClient? 

It would not be feasible to have a "create_transfer_request" which works for all external APIs as they will most likely require different requests.

Reply:
@ChrOertlin Yes, precisely 😃